File: python/sieve.py
import sys, math

def main(input_value: str):
    if validate(input_value):
        print('Prime List:', sieve(int(input_value)))
        return
    print("Use an integer number (2 or more) as an argument.")
    return

def validate(data: str) -> bool:
    try:
        int_data: int = int(data)
    except ValueError:
        return False
    if int_data < 2:
        return False
    return True

def sieve(count: int) -> list:
    natural: list = [True] * (count + 1)
    natural[1] = False
    prime: list = []
    
    i: int = 2
    while i < math.sqrt(count):
        if natural[i]:
            prime.append(i)
            for j in range(i**2, count+1, i):
                natural[j] = False
        i += 1
    for k in range(i, count+1):
        if natural[k]:
            prime.append(k)
    return prime

if __name__=='__main__':
    main(sys.argv[1])

File: TypeScript/src/ts/sieve.ts
export const sieve = (count: int): number[] => {
    const natural: boolean[] = Array(count+1).fill(true).fill(false, 0, 2);
    const prime: boolean[] = [];

    for (var i = 2; i < Math.sqrt(count); i++) {
        if (natural[i]) {
            prime.push(i);
            for (let j = i**2; j <= (count+1); j+=i) {
                natural[j] = false;
            }
        }
    }
    for (let k = i; k <= (count+1); k++) {
        if (natural[k]) {
            prime.push(k);
        }
    }
    return prime;
};